posthog/models/error_tracking/error_tracking.py

```diff
@@ -0,0 +1,25 @@
+from django.db import models
+
+
+class ErrorTrackingGroup(models.Model):
+    class Status(models.TextChoices):
+        ARCHIVED = "archived", "Archived"
+        ACTIVE = "active", "Active"
+        RESOLVED = "resolved", "Resolved"
+        PENDING_RELEASE = "pending_release", "Pending release"
+
+    team: models.ForeignKey = models.ForeignKey("Team", on_delete=models.CASCADE)
+    created_at: models.DateTimeField = models.DateTimeField(auto_now_add=True, blank=True)
+    assignee: models.ForeignKey = models.ForeignKey(
+        "User",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+    )
+
+    status: models.CharField = models.CharField(max_length=40, choices=Status.choices, default=Status.ACTIVE)
+
+
+class ErrorTrackingFingerprint(models.Model):
```

Comment:
Went with a separate model here rather than an `ArrayField` within the `ErrorTrackingGroup` because _the internet_ said that searching arrays in Postgres wasn't the most efficient

Given an array of `fingerprints` should only be 1 to ~5 merged values, maybe it just makes sense to go with a single model

Reply:
i really don't enjoy django's orm and the `insight` <-> `dashboard tile` <-> `dashboard` relation is a pain to manage so i'd definitely go with a single model here

we can always build a reverse lookup if we need to speed something up

so I was picturing we'd have

ErrorTrackingGroup
---
fingerprint -> string[]
status -> TextChoice
assignee -> FK[user]
merged_fingerprints -> string[][]  * this could be pipe separated list of comma separated strings as a sting and we search needle in haystack
is_ignored -> boolean -> we'll hold these in memory in plugin server (at least for a simple implementation)
archived -> boolean -> we'll hold these in memory in plugin server (at least for a simple implementation)

difference between ignored and archived (and come up with different names by all means e.g. should_drop and something_else) is that one we drop at ingestion, no billing, no nothing, the other we continue to ingest but we don't alert you about them when they happen

Reply:
at least at first the majority of groups will have one fingerprint and no merged fingerprints
so i'd optimise for that

plus we'll only need to store a group when we move away from the default
e.g. all groups are implicityly unassigned and unresolved and not ignored and not archived and not merged

🤔 which maybe means the likelihood of merged fingerprints per stored group is higher since you're only storing interesting ones

maybe write a couple of tests on the lifecycles of the model and see how it feels like this, see where it takes you

we can add indexes for the likely queries then too

Reply:
we can inherit from UUIDModel and get a non int id too

Reply:
`archived` and `ignored` make sense to me as names, I like when it closely aligns with what it means from a users perspective.

Is it necessary to have those two states called out separately to the `status`? I know we'll need them in the plugin server but maybe we could just cache the `status` there instead.

The only other thing I'm thinking about is the case where we want to ignore for a certain period of time e.g. until the error occurs another 1000 times. I _think_ that would be covered by the archived case (with some special conditions to be defined in future).

Ignoring seems like a separate action that you would use much more rarely but probably still helpful to have in some cases to save on event volume

Reply:
Chatted through this in person. Archived can live as a status on the model. We may need an extra column in future if we want to allow people to add archive rules e.g. until 100 more users are affected.

Ignored feels like a separate concept. It is for cases when you do not want to ingest the error anymore. We're not quite sure whether customers will want to do that at a fingerprint (e.g. group) level or using regex / property expressions. Going to punt on it until we have a better sense once someone asks for it after releasing a beta